pkg/lib/authenticationflow/declarative/intent_account_linking_oauth.go

```diff
@@ -0,0 +1,129 @@
+	case 1: // Add NodeDoCreateIdentity
+		return nil, nil
+	}
+
+	return nil, authflow.ErrEOF
+}
+
+func (i *IntentAccountLinkingOAuth) ReactTo(ctx context.Context, deps *authflow.Dependencies, flows authflow.Flows, input authflow.Input) (*authflow.Node, error) {
+	flowUserID, err := getUserID(flows)
+	conflictedUserID := i.ConflictingIdentity.UserID
+	if err != nil {
+		return nil, err
+	}
+	if flowUserID != conflictedUserID {
+		err = authflow.TraverseFlow(authflow.Traverser{
+			NodeSimple: func(nodeSimple authflow.NodeSimple, w *authflow.Flow) error {
+				milestone, ok := nodeSimple.(MilestoneSwitchToExistingUser)
+				if ok {
+					err = milestone.MilestoneSwitchToExistingUser(deps, w, conflictedUserID)
+					if err != nil {
+						return err
+					}
+				}
+				return nil
+			},
+			Intent: func(intent authflow.Intent, w *authflow.Flow) error {
+				milestone, ok := intent.(MilestoneSwitchToExistingUser)
+				if ok {
+					err = milestone.MilestoneSwitchToExistingUser(deps, w, conflictedUserID)
+					if err != nil {
+						return err
+					}
+				}
+				return nil
+			},
+		}, flows.Root)
+		if err != nil {
+			return nil, err
+		}
+		// Use synthetic input to auto select the conflicted identity in the login flow
+		return nil, &authflow.ErrorRewriteFlow{
+			Intent:         flows.Root.Intent,
+			Nodes:          flows.Root.Nodes,
+			SyntheticInput: i.createSyntheticInputOAuthConflict(i.OAuthIdentitySpec, i.ConflictingIdentity),
+		}
+	}
+
+	switch len(flows.Nearest.Nodes) {
+	case 0:
+		// TODO(tung): Check the config to decide what to do
+		flowReference := authflow.FlowReference{
+			Type: authflow.FlowTypeLogin,
+			// FIXME(tung): This should be read from config
+			Name: "default",
+		}
+		loginIntent := IntentLoginFlow{
+			TargetUserID:  conflictedUserID,
+			FlowReference: flowReference,
+		}
+		return authflow.NewSubFlow(&loginIntent), nil
+	case 1:
+		info, _, err := newIdentityInfo(deps, i.LinkToUserID, i.OAuthIdentitySpec)
+		if err != nil {
+			return nil, err
+		}
+		return authflow.NewNodeSimple(&NodeDoCreateIdentity{
+			Identity: info,
+		}), nil
+	}
+
+	return nil, authflow.ErrIncompatibleInput
+}
+
+func (n *IntentAccountLinkingOAuth) createSyntheticInputOAuthConflict(oauthIden *identity.Spec, conflictedInfo *identity.Info) *SyntheticInputOAuthConflict {
+	input := &SyntheticInputOAuthConflict{}
+
+	switch conflictedInfo.Type {
+	case model.IdentityTypeLoginID:
+		input.LoginID = conflictedInfo.LoginID.LoginID
+		switch conflictedInfo.LoginID.LoginIDType {
+		case model.LoginIDKeyTypeEmail:
+			input.Identification = config.AuthenticationFlowIdentificationEmail
+		case model.LoginIDKeyTypePhone:
+			input.Identification = config.AuthenticationFlowIdentificationPhone
+		case model.LoginIDKeyTypeUsername:
+			input.Identification = config.AuthenticationFlowIdentificationUsername
+		}
+	case model.IdentityTypeOAuth:
+		input.Identification = config.AuthenticationFlowIdentificationOAuth
+	case model.IdentityTypePasskey:
```

Comment:
Theoretically, it is impossible to conflict with a passkey identity. Remove this case?